apps/history/declarativehistorymodel.cpp

```diff
@@ -57,17 +62,26 @@ void DeclarativeHistoryModel::remove(int index)
 
 void DeclarativeHistoryModel::remove(const QString &url)
 {
+    const QString sanitizedUrl = FaviconManager::sanitizedHostname(url);
+    bool canClearFavicon = true;
     int index = 0;
     for (const auto &link : m_links) {
         if (link.url() == url) {
             remove(index);
-            return;
+        } else if (FaviconManager::sanitizedHostname(link.url()) == sanitizedUrl) {
+            // this site is still represented in history, so cannot clear icon
+            canClearFavicon = false;
         }
         index++;
     }
 
     // Not in model but remove from database anyway
     DBManager::instance()->removeHistoryEntry(url);
```

Comment:
In most cases `DBManager::instance()->removeHistoryEntry(url)` will now be called twice for the same entry (once here, and once in the call  to `remove(index)` on line 70. Does this matter?

Just a very small point, but the comment no longer makes sense, so probably better to remove it to avoid confusion.

Reply:
I assumed the point of this was to catch the case where the url wasn't in the model (for some reason) but was in the database.

Reply:
Yes, that was my assumption too. I was just being over picky here, because now this line will get called even if the entry *is* in the model. But still the code comment makes sense anyway to clarify things, so please ignore my suggestion.